File: 프로그래머스/1/12982. 예산/예산.cpp
#include <iostream>
#include <stdio.h>
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

int solution(vector<int> d, int budget) {
    int answer = 0;
    int res = 0;
    sort(d.begin(),d.end());
    
    for(auto q : d)
    {
        if(res + q <= budget)
        {
            res += q;
            answer ++;
        }
        else break;
    }
    return answer;
}

File: 프로그래머스/1/12982. 예산/예산.cs
using System;

public class Solution {
    public int solution(int[] d, int budget) {
        int answer = 0;
        Array.Sort(d);
        int sum = 0;
        foreach(int i in d)
        {
            sum+=i;
            if(budget >= sum) answer++;
        }
        return answer;
    }
}